### PR TITLE
WIP: doc updates

### PR DESCRIPTION
**Pull request by winder**
Additional documentation about discovery. New documentation for consensus and component links.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -39,41 +35,184 @@ addresses.
...
+    bytes[] signers; // The onchain signing keys of nodes in the don.
+    bytes[] transmitters; // The onchain transmitter keys of nodes in the don.
+    bytes offchainConfig; // The offchain configuration for the OCR3 protocol. Protobuf encoded.
+  }
+}
+```
+* [offchainConfig](https://github.com/smartcontractkit/chainlink/blob/200fb4504b6b96c92979fe42148530f784098cc2/core/capabilities/ccip/oraclecreator/plugin.go#L526) has two components:
+  * [commitOffchainConfig](https://github.com/smartcontractkit/chainlink-ccip/blob/ae3e8f4935a0bb3efe86c456b410a612be56bfec/pluginconfig/commit.go#L77)
+  * [execOffchainConfig](https://github.com/smartcontractkit/chainlink-ccip/blob/ae3e8f4935a0bb3efe86c456b410a612be56bfec/pluginconfig/execute.go#L14)
+```
```
Comment (makramkd) [inline docs/configuration.md:60]: Language tag 
```suggestion
```golang
```

File: docs/configuration.md [hunk 2]
```diff
@@ -83,20 +222,33 @@ all chains. For example the OffRamp is only available to nodes configured with
...
+OffRamp ->> DestReader: Get dynamic configuration.
+OffRamp ->> DestReader: Get source chain configuration.
 DestReader ->> DON: Share contract addresses.
-DON ->> SourceReader(s): Receive contract addresses.
-SourceChain(OnRamp) ->> SourceReader(s): Receive OnRamp configuration
+OnRamp ->> SourceReader(s): Get static configuration.
+OnRamp ->> SourceReader(s): Get dynamic configuration.
+OnRamp ->> SourceReader(s): Get source chain configuration.
+SourceReader(s) ->> DON: Share contract addresses.
 ```
```
Comment (makramkd) [inline docs/configuration.md:240]: I think this mermaid diagram is an excellent resource but I wonder if the arrow directions should be reversed - I'm understanding the arrows here basically as abstract "function calls", e.g

```mermaid
sequenceDiagram
OnRamp ->> SourceReader(s): Get static configuration.
```

I'm reading this as "OnRamp calls SourceReader to get the static configuration" but I feel like it should be the other way around:

```mermaid
sequenceDiagram
SourceReader(s) ->> OnRamp(s): Get static configuration.
```

File: docs/consensus.md [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
+
+## Configuration
+
+Because OCR is a permissioned network, roles are assigned during configuration.
+The plugin's must ensure observations for each chain are only made by the
+designated nodes.
+
+Each role also has its own value for `FChain`. This value determines the number
+of nodes which must be assigned the role and the threshold of observations
+required.
```
[the remark this answers is not in the document]
Comment (winder) [reply docs/consensus.md:36]: Now that I'm looking at this, it occurs to me that we never assert 3F+1 or 3FChain+1. Only the thresholds

File: docs/configuration.md (hunk 2 above)
Comment (winder) [reply docs/configuration.md]: Improved this with notes and the full call/response

image

**Review by makramkd — APPROVED** [review]

**Pull request merged**